### PR TITLE
Allocate real blockSize when loading files for distributedLoad

### PR DESCRIPTION
**Pull request by apc999**
In issue #13005, `distributedLoad` failed due to reserving too much space (e.g., 128MB space reserved just for loading 300KB files). 
When write concurrency is high, it can take a lot of wasted space and possibly causing `distributedLoad` failed.

This PR
1. makes better estimation on DataWriter to reserve bytes 
2. optimizes the performance of `distributedLoad` by reducing master RPCs by moving `getStatus` call out of loop

Fix https://github.com/Alluxio/alluxio/issues/13005

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java [hunk 1]
```diff
@@ -251,29 +249,21 @@ public BlockInStream getInStream(BlockInfo info, InStreamOptions options,
    */
   public BlockOutStream getOutStream(long blockId, long blockSize, WorkerNetAddress address,
       OutStreamOptions options) throws IOException {
-    if (blockSize == -1) {
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java:254]: I checked the src code, blockSize will never be -1 after this PR.
using blockSize as a switch to indicate existence is anyway a bad practise.

File: core/client/fs/src/main/java/alluxio/client/block/stream/DataWriter.java [hunk 2]
```diff
@@ -72,7 +72,7 @@ public static DataWriter create(FileSystemContext context, long blockId, long bl
               context, address, blockId, blockSize, options);
         }
         LOG.debug("Creating short circuit output stream for block {} @ {}", blockId, address);
-        return LocalFileDataWriter.create(context, address, blockId, options);
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/block/stream/DataWriter.java:75]: `LocalFileDataWriter.create` was not taking advantage of block size information

File: job/server/src/main/java/alluxio/job/util/JobUtils.java [hunk 3]
```diff
@@ -125,15 +123,10 @@ public static void loadBlock(FileSystem fs, FileSystemContext context, String pa
           .getMessage(blockId));
     }
 
-    // TODO(jiri): Replace with internal client that uses file ID once the internal client is
-    // factored out of the core server module. The reason to prefer using file ID for this job is
-    // to avoid the the race between "replicate" and "rename", so that even a file to replicate is
-    // renamed, the job is still working on the correct file.
-    URIStatus status = fs.getStatus(new AlluxioURI(path));
```
Comment (apc999) [inline job/server/src/main/java/alluxio/job/util/JobUtils.java:132]: take RPC out of the loop

File: job/server/src/main/java/alluxio/job/util/JobUtils.java [hunk 4]
```diff
@@ -154,8 +147,11 @@ public static void loadBlock(FileSystem fs, FileSystemContext context, String pa
         LocalFirstPolicy.class.getCanonicalName(), conf));
 
     // use -1 to reuse the existing block size for this block
+    BlockInfo blockInfo = status.getBlockInfo(blockId);
+    Preconditions.checkNotNull(blockInfo, "Can not find block %s in status %s", blockId, status);
+    long blockSize = blockInfo.getLength();
     try (OutputStream outputStream =
-        blockStore.getOutStream(blockId, -1, localNetAddress, outOptions)) {
```
Comment (apc999) [inline job/server/src/main/java/alluxio/job/util/JobUtils.java:158]: we know blockSize, just fill it, using -1 is very confusing

File: job/server/src/main/java/alluxio/job/plan/load/LoadDefinition.java [hunk 5]
```diff
@@ -138,9 +139,16 @@ public LoadDefinition() {
...
+    // TODO(jiri): Replace with internal client that uses file ID once the internal client is
+    // factored out of the core server module. The reason to prefer using file ID for this job is
+    // to avoid the the race between "replicate" and "rename", so that even a file to replicate is
+    // renamed, the job is still working on the correct file.
+    URIStatus status = context.getFileSystem().getStatus(new AlluxioURI(config.getFilePath()));
+
     for (LoadTask task : tasks) {
       JobUtils.loadBlock(
-          context.getFileSystem(), context.getFsContext(), config.getFilePath(), task.getBlockId());
+          status, context.getFsContext(), task.getBlockId());
```
Comment (calvinjia) [inline job/server/src/main/java/alluxio/job/plan/load/LoadDefinition.java:151]: One line

File: job/server/src/main/java/alluxio/job/util/JobUtils.java [hunk 6]
```diff
@@ -154,8 +147,11 @@ public static void loadBlock(FileSystem fs, FileSystemContext context, String pa
         LocalFirstPolicy.class.getCanonicalName(), conf));
 
     // use -1 to reuse the existing block size for this block
+    BlockInfo blockInfo = status.getBlockInfo(blockId);
```
Comment (calvinjia) [inline job/server/src/main/java/alluxio/job/util/JobUtils.java:150]: Remove the comment above?

**Review by calvinjia — CHANGES_REQUESTED** [review]
Thanks for cleaning this up!

**Review by calvinjia — APPROVED** [review]

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**